include/exadg/postprocessor/normal_flux_calculation.cpp

```diff
@@ -0,0 +1,179 @@
+/*  ______________________________________________________________________
+ *
+ *  ExaDG - High-Order Discontinuous Galerkin for the Exa-Scale
+ *
+ *  Copyright (C) 2021 by the ExaDG authors
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *  ______________________________________________________________________
+ */
+
+// C/C++
+#include <fstream>
+
+// ExaDG
+#include <exadg/postprocessor/normal_flux_calculation.h>
+#include <exadg/utilities/create_directories.h>
+
+namespace ExaDG
+{
+template<int dim, typename Number>
+NormalFluxCalculator<dim, Number>::NormalFluxCalculator(
+  dealii::MatrixFree<dim, Number> const & matrix_free_in,
+  unsigned int const                      dof_index_in,
+  unsigned int const                      quad_index_in,
+  NormalFluxCalculatorData const &        data_in,
+  MPI_Comm const &                        mpi_comm_in)
+  : matrix_free(matrix_free_in),
+    dof_index(dof_index_in),
+    quad_index(quad_index_in),
+    clear_files(true),
+    mpi_comm(mpi_comm_in)
+{
+  data = data_in;
+
+  for(auto it : data.boundary_ids)
+    flux.insert(typename std::pair<dealii::types::boundary_id, double>(it, 0.0));
+
+  if(data.evaluate)
+    create_directories(data.directory, mpi_comm);
+}
+
+template<int dim, typename Number>
+void
+NormalFluxCalculator<dim, Number>::evaluate(VectorType const & solution,
+                                            double const       time,
+                                            bool const         unsteady)
+{
+  std::cout << "Evaluate_normal_flux" << std::endl;
+
+  // zero values since we sum into these variables
+  for(auto & iterator : flux)
+  {
+    iterator.second = 0.0;
+  }
+
+  FaceIntegratorScalar integrator(matrix_free, true, dof_index, quad_index);
+
+  std::map<dealii::types::boundary_id, double> area(flux);
+
+  for(unsigned int face = matrix_free.n_inner_face_batches();
+      face < (matrix_free.n_inner_face_batches() + matrix_free.n_boundary_face_batches());
+      face++)
+  {
+    typename std::map<dealii::types::boundary_id, double>::iterator it;
+    dealii::types::boundary_id boundary_id = matrix_free.get_boundary_id(face);
+
+    it = flux.find(boundary_id);
```

Comment:
Wouldn't it be nicer to remove the longish variable definition three lines above by making use of the `auto` keyword?
```suggestion
    auto it = flux.find(boundary_id);
```

Reply:
agree